lib/connection_pool.rb

```diff
@@ -44,6 +44,32 @@ def self.wrap(options, &block)
     Wrapper.new(options, &block)
   end
 
+  def self.after_fork
+    ObjectSpace.each_object(ConnectionPool) do |pool|
+      # We're on after fork, so we know all other threads are dead.
+      # All we need to do is to ensure the main thread doesn't have a
+      # checked out connection
+      pool.checkin(force: true)
+
+      # Unfortunately we don't know what method to call to close the connection.
+      # So we simply drop the reference and let Ruby close the sockets.
+      pool.reload { |_c| }
```

Comment:
Use `c.close if c.respond_to?(:close)`?

Reply:
I didn't want to assume that, but that is for sure the most common convention. I'll add it.